packages/jest-util/src/fake_timers.js

```diff
@@ -486,7 +486,7 @@ export default class FakeTimers<TimerRef> {
       return null;
     }
 
-    if (delay == null) {
+    if (delay == null || Number.isNaN(delay) || !Number.isFinite(delay)) {
```

Comment:
You can cover all cases in one shot:
```js
delay = Number(delay) | 0;
```

This invokes the same conversion algorithm as in the WebIDL spec. `Number(x)` coerces anything to a number (or `NaN`), and `x | 0` truncates it to a 32-bit signed integer, with `NaN` and `Infinity` converted to `0`.

Reply:
tnx for the suggestion its indeed more straight forward and covers also more cases, updated 14881d77e0056bd54265a3f9ffc2710c4fb58ac9